### PR TITLE
Add BlastInputSequences to BlastForm

### PR DESCRIPTION
**Pull request by azangru**
## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1466
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1468

## Description
- add initial grid to the BlastForm page
- add redux slice to control the BlastForm page
- add BlastInputSequences to the BlastForm page
- remove the BlastInputSequences story from Storybook

## Deployment URL
http://add-blast-sequence-to-page.review.ensembl.org

### CODE REVIEW

File: src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequences.scss [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding-left: 28px;
+  padding-bottom: 12px;
+}
+
+// .headerCompact {
+//   padding-right: 0;
+// }
```
Comment (azangru) [inline src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequences.scss:18]: Note to self: delete the commented code.

File: src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequences.test.tsx [hunk 2]
```diff
@@ -15,35 +15,56 @@
...
-
-    it('has a control to clear all sequences', () => {
-      const { container } = render(<BlastInputSequences />);
-      const clearAll = container.querySelector('.header .clearAll');
-      expect(clearAll).toBeTruthy();
-    });
-  });
-
   describe('with added sequences', () => {
     // TODO: tests to be written after the component is connected to redux
```
Comment (imransl) [inline src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequences.test.tsx:69]: Are tests going to be added here now that the component is connected to redux?

File: src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequencesHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+type AddAnotherSequenceProps = {
+  disabled?: boolean;
+  onAdd: () => void;
+};
+
+const AddAnotherSequence = (props: AddAnotherSequenceProps) => {
+  return (
+    <div className={styles.addSequence}>
+      <span className={styles.addSequenceLabel}>Add another sequence</span>
+      <PlusButton onClick={props.onAdd} disabled={props.disabled} />
```
Comment (imransl) [inline src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequencesHeader.tsx:100]: Can't the props be just passed as `{...props}` if `onAdd` is changed to `onClick`?

File: src/content/app/tools/blast/views/blast-form/BlastForm.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import { getStep } from 'src/content/app/tools/blast/state/blast-form/blastFormSelectors';
+
+import ToolsAppBar from 'src/content/app/tools/shared/components/tools-app-bar/ToolsAppBar';
+import ToolsTopBar from 'src/content/app/tools/shared/components/tools-top-bar/ToolsTopBar';
+
+import BlastInputSequencesHeader from 'src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequencesHeader';
+import BlastInputSequences from 'src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequences';
+
+import styles from './BlastForm.scss';
+import { useSelector } from 'react-redux';
```
Comment (imransl) [inline src/content/app/tools/blast/views/blast-form/BlastForm.tsx:30]: Import order?

File: src/shared/hooks/useMediaQuery.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    mediaQueryList?.addEventListener('change', onChange);
+
+    return () => mediaQueryList?.removeEventListener('change', onChange);
+  }, [mediaQuery]);
+
+  return doesMatch;
+};
+
+const getMediaQueryList = (mediaQuery: string) => {
+  if (!isClient()) {
```
Comment (imransl) [inline src/shared/hooks/useMediaQuery.ts:39]: Wouldn't the inverse of this check be neater?

File: src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequencesHeader.tsx (hunk 3 above)
Comment (azangru) [reply src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequencesHeader.tsx:100]: They could; but re-reading this code, I think I still prefer `onAdd` to `onClick` as a property name for the `AddAnotherSequence` component. `PlusButton` is definitely clicked; whereas `AddAnotherSequence` has a clickable and a non-clickable part.

File: src/content/app/tools/blast/views/blast-form/BlastForm.tsx (hunk 4 above)
Comment (azangru) [reply src/content/app/tools/blast/views/blast-form/BlastForm.tsx]: Must have been VSCode 😕 Fixed.

File: src/shared/hooks/useMediaQuery.ts (hunk 5 above)
Comment (azangru) [reply src/shared/hooks/useMediaQuery.ts]: Much neater. Updated.

File: src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequences.test.tsx (hunk 2 above)
Comment (azangru) [reply src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequences.test.tsx]: Added tests.

**Review by jyothishnt — COMMENTED** [review]
Looks good. Few things that I have noticed are listed below

- Not limiting to 30 sequences
- Workflow => paste a sequence,  click on "Add another sequence".  Click on the first input box and then clicking on the second box removes the second box. Felt weird though we would like to remove all empty boxes on blur.
- On adding new sequence should we move the focus to the new inputbox?

**Comment by imransl** [comment]
@azangru I am happy with the tests. @jyothishnt has made some suggestions so we can merge after they are addressed?

**Pull request merged**